crates/kornia-tensor/src/allocator.rs

```diff
@@ -30,17 +31,36 @@ pub trait TensorAllocator: Clone {
     fn alloc(&self, layout: Layout) -> Result<*mut u8, TensorAllocatorError>;
 
     /// Deallocates memory for a tensor with the given layout.
+    ///
+    /// # Arguments
+    ///
+    /// * `ptr` - A non-null pointer to the allocated memory.
+    /// * `layout` - The layout of the tensor.
     fn dealloc(&self, ptr: *mut u8, layout: Layout);
 }
 
 #[derive(Clone)]
 /// A tensor allocator that uses the system allocator.
-pub struct CpuAllocator;
+pub struct CpuAllocator {
+    parent: Option<Arc<dyn ParentDeallocator>>,
```

Comment:
Do we have benchmarks before/after as this adds: 1 branch to get the option, 1 sync with Arc, 1 virtual table access?

Reply:
I feel adding the parent in this struct it’s unnecessary. The allocators are generics in the tensor storage which you can specialise (eg adding the gst buffer or anything) as soon as you implement the trait methods. I’d explore going to that route